File: LCM.py
"""
    Python Program to calculate the LCM of two numbers
"""
def LCM(num1, num2):
    greater = max(num1, num2)
    smaller = min(num1, num2)
    HCF = 0

    # Finding HCF
    for i in range(1, smaller+1):
        if(greater%i == 0) and (smaller%i == 0):
            HCF = i
    
    # Finding LCM
    if (num1%num2 == 0) and (num2%num1 == 0):
        return greater
    else: 
        return int((num1*num2)/HCF)

print("Enter the two numbers whose LCM is needed: ")
a = int(input())
b = int(input())

print(f"The LCM of {a} and {b} is: {LCM(a, b)}")

File: LCM.cpp
/*
 *	CPP Program to print the LCM of two numbers
 */
#include <iostream>
int main(){
	int a, b;
	std::cout << "Enter the two numbers: ";
	std::cin >> a >> b;
	
	//Finding the HCF of the two numbers:
	int greater, smaller;

	if(a > b){
		greater = a;
		smaller = b;
	}
	else{
		greater = b;
		smaller = a;
	}
	
	// FINDING THE LCM OF THE TWO NUMBERS
	int HCF;
	for(int i = 1; i < greater ; i++){
		if((smaller % i == 0 ) && (greater%i == 0)){
			HCF = i;
		}
	}
	

	// FINALLY FINDING THE LCM
	int LCM;
	if (greater % smaller == 0){
		LCM = greater;
	}
	else{
		LCM = (greater*smaller)/HCF;
	}
	std::cout << "The LCM of " << a << " and " << b << " is: " << LCM << std::endl;

}




